### DISCUSSION TITLE
Setting up language server for Sublime Text

### DISCUSSION THREAD

**Discussion by niksy** [oxc-project/oxc#5061] [Q&A]
I’m trying to enable language server inside Sublime Text.

I’ve managed to create basic configuration. Everything works if default settings are provided, but if I try to change `configPath` to some different config file (e.g. `oxlintrc.json`), language server doesn’t pick this up.

These are default options I’m using:

```json
{
  "oxc_language_server.configPath": ".eslintrc",
  "oxc_language_server.enable": true,
  "oxc_language_server.run": "onType",
  "oxc_language_server.trace.server": "off"
}
```

And corresponding Sublime LSP configuration:

```json
{
  "LSP-oxlint": {
    "enabled": true,
    "command": [
      "node", "/dev/node_modules/oxlint/bin/oxc_language_server"
    ],
    "settings": {
      "oxc_language_server.configPath": "oxlintrc.json",
      "oxc_language_server.enable": true,
      "oxc_language_server.run": "onType",
      "oxc_language_server.trace.server": "off"
    },
    "selector": "source.js"
  }
}

```

If I change some other option, everything is okay—for example, setting `oxc_language_server.enable` to false disables language server.

Am I doing something wrong? From what I can see inside [language server codebase](https://github.com/oxc-project/oxc/blob/4e1f4abf8979fa2bb81543481904908309ab684c/crates/oxc_language_server/src/main.rs#L340) this should be picked up and applied inside current workspace.

**Comment by Sysix** [comment]
Hello,

this was a bug and fixed in https://github.com/oxc-project/oxc/pull/7447

The newest release this should be fixed :)

**Comment by niksy** [comment]
Has this been released in latest oxlint version? Because it seems like this is still an issue.

**Reply by Sysix** [reply]
it should be. How do you get your binary file?

**Reply by niksy** [reply]
> it should be. How do you get your binary file?

From globally installed oxlint, version 0.14.0.

E.g. `/global_node_modules/bin/oxc_language_server`.

**Reply by Sysix** [reply]
I thinks you have an old prefix. The new one is  `oxc.*` instead of `oxc_language_server.*`. 

**When it is still not working:**
I do not really know Sublime Text. Are you fimilar with [LSP](https://microsoft.github.io/language-server-protocol/)?
Does Sublime Text send a `workspace/didChangeConfiguration` when you changing your config? 

You can see what the LSP Server supports at the moment in the [README.md](https://github.com/oxc-project/oxc/blob/main/crates/oxc_language_server/README.md)

**Reply by niksy** [reply]
Yes, Sublime Text has package for LSP support: https://lsp.sublimetext.io/

This is my attempt in creating LSP client for Oxc language server.

> I thinks you have an old prefix. The new one is oxc.* instead of oxc_language_server.*.

I think this needs to be left as-is, looking at this line settings is parsed with this key: https://github.com/oxc-project/oxc/blob/245d7d922c123aa86bd545c0dbd5e379abda677d/crates/oxc_language_server/src/main.rs#L154

If I change it to `oxc`, settings are not picked up.

> Does Sublime Text send a workspace/didChangeConfiguration when you changing your config?

It seems like settings are not picked up if Oxlint is used globally. If I create Sublime Text project (which is workspace concept in ST), settings are picked up. Is there some part of code that doesn’t handle this without workspace?

Also, changing config where rules are defined doesn’t seem to be picked up in `workspace/didChangeWatchedFiles`.

**Reply by Sysix** [reply]
> This is my attempt in creating LSP client for Oxc language server

Thanks you for doing this ❤️ 

> Is there some part of code that doesn’t handle this without workspace?

I think the server is currently designed to work only with workspaces.

> Also, changing config where rules are defined doesn’t seem to be picked up in workspace/didChangeWatchedFiles.

~~For oxlint config changes, you need to send a [workspace/didChangeWatchedFiles](https://github.com/oxc-project/oxc/tree/main/crates/oxc_language_server#workspacedidchangewatchedfiles) event~~

EDIT: make sure you define `oxc_language_server.configPath` or `oxlint` will fallback for `.oxlintrc.json` in the current workspace. When nothing is found. If will fallback to the default settings
